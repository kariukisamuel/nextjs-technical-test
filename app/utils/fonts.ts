import { Montserrat, Roboto } from "next/font/google"

export const montserrat_init = Montserrat({
    variable: '--font-montserrat',
    weight: '700',
    subsets:['latin']
});
export const roboto_init = Roboto({
    variable: '--font-roboto',
    weight: '300',
    subsets:['latin']
})

export const montserrat = montserrat_init.variable;
export const roboto = roboto_init.variable;