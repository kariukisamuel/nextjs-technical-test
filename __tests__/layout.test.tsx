import React from "react";
import { render } from "@testing-library/react";
import RootLayout from "@/app/layout";
import { montserrat, roboto } from "@/app/utils/fonts";

jest.mock("../app/components/Common/GoogleAnalytics", () => () => <div>Google Analytics</div>);
jest.mock("../app/components/Common/SchemaMarkup", () => () => <div>Schema Markup</div>);

describe("RootLayout", () => {
  it("renders children correctly", () => {
    const { getByText } = render(
      <RootLayout>
        <h1>Hello World</h1>
      </RootLayout>
    );
    
    expect(getByText(/hello world/i)).toBeInTheDocument();
  });

  it("includes SchemaMarkup and GoogleAnalytics components", () => {
    const { getByText } = render(
      <RootLayout>
        <h1>Hello World</h1>
      </RootLayout>
    );

    expect(getByText("Schema Markup")).toBeInTheDocument();
    expect(getByText("Google Analytics")).toBeInTheDocument();
  });

  it("applies the correct classes to the body", () => {
    const { container } = render(
      <RootLayout>
        <h1>Hello World</h1>
      </RootLayout>
    );

  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <RootLayout>
        <h1>Hello World</h1>
      </RootLayout>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
