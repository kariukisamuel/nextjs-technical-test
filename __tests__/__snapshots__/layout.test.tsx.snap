// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    Schema Markup
  </div>
  <h1>
    Hello World
  </h1>
  <div>
    Google Analytics
  </div>
</DocumentFragment>
`;
